test/local_integration_test.py

```diff
@@ -0,0 +1,182 @@
+    bundle_uuid_prefix = None
+    prefix_length = 3
+
+    def setUp(self):
+        super().setUp()
+        self.bundle_uuid_prefix = ''.join([str(random.choice('abcdef0123456789')) for _ in range(self.prefix_length)])
+        plugin = Plugin.load()
+        test_dss_query = copy.deepcopy(plugin.dss_subscription_query())
+        prefix_section = {
+            "prefix": {
+                "uuid": self.bundle_uuid_prefix
+            }
+        }
+        test_dss_query['query']['bool']['must'].append(prefix_section)
+        self.dss_query = test_dss_query
+
+    def tearDown(self):
+        subscribe(False)
+
+    def test_webservice_and_indexer(self):
+        subscribe(True)
+        query = self.dss_query
+        test_uuid = str(uuid.uuid4())
+        test_name = f'integration-test_{test_uuid}_{self.bundle_uuid_prefix}'
+        test_reindexer = Reindexer(indexer_url=config.indexer_endpoint(),
+                                   test_name=test_name,
+                                   es_query=query)
+
+        if False:
+            logger.info('Deleting all indices ...')
+            test_reindexer.delete_all_indices()
+
+        logger.info('Starting test using test name, %s ...', test_name)
+        logger.info('Creating indices and reindexing ...')
+        selected_bundle_fqids = test_reindexer.reindex()
+        self.num_of_bundles = len(selected_bundle_fqids)
+        self.check_bundles_are_indexed(test_name, 'files', set(selected_bundle_fqids))
+        self.check_endpoint_is_working('/')
+        self.check_endpoint_is_working('/health')
+        self.check_endpoint_is_working('/version')
+        self.check_endpoint_is_working('/repository/summary')
+        self.check_endpoint_is_working('/repository/files/order')
+        self.check_endpoint_is_working('/repository/files/export')
+
+    def check_endpoint_is_working(self, url: str):
+        url = config.service_endpoint() + url
+        response = requests.get(url)
+        response.raise_for_status()
+
+    def get_number_of_messages(self, queues: Dict[str, boto3.resources.base.ServiceResource]):
+        total_message_count = 0
+        for queue_name, queue in queues.items():
+            attributes = tuple('ApproximateNumberOfMessages' + suffix for suffix in ('', 'NotVisible', 'Delayed'))
+            num_messages = tuple(int(queue.attributes[attribute]) for attribute in attributes)
+            num_available, num_inflight, num_delayed = num_messages
+            queue.reload()
+            message_sum = sum((num_available, num_inflight, num_delayed))
+            logger.info('Queue %s has %i messages, %i available, %i in flight and %i delayed',
+                        queue_name, message_sum, num_available, num_inflight, num_delayed)
+            total_message_count += message_sum
+        logger.info('Total # of messages: %i', total_message_count)
+        return total_message_count
+
+    @property
+    def default_queue_empty_timeout(self):
```

Comment:
It would be `num_bundles` but I don't like stashing state like that as a hidden communication channel between methods. We talked about this before: mutable state is better passed explicitly than stashed as instance state, except when passing it around is tedious. If the timeout is a function of the number of bundles, the most obvious choice would be to make that a parameter to a method that returns the timeout.